src/test/java/ca/bc/gov/ag/efax/ws/service/DocumentDistributionServiceTest.java

```diff
@@ -85,6 +96,70 @@ public void testReceiveMustBeAFax() throws DatatypeConfigurationException {
         assertEquals(0, emails.size());
     }
 
+    @Test
+    public void testInitiateFaxSendFault() throws Exception {
+        // Attempt to send a message to a broken or offline exchange server - should result in a FAXSendFault
+
+        // exchangeService - add the email to the List<EmailMessage> when "sending" an email.
+        doAnswer(invocation -> {
+            throw new FAXSendFault("Offline");
+        }).when(exchangeService).sendItem(any(), any());
+        
+        DocumentDistributionRequest request = createBaseRequest();
+        try {
+            documentDistributionService.initiateRequestToSendMessage(request);
+            fail("Expected FAXSendFault since ExchangeServer is offline.");
+        } catch (FAXSendFault e) {
+        } catch (Exception e) {
+            fail("Unexpected Exception thrown.", e);
+        } 
+
+        // assert no emails sent
+        assertEquals(0, emails.size());
+    }
+
+    @Test
+    public void testSendResponseToCallback_valid() {
+        DocumentDistributionMainProcessProcessResponse response = new DocumentDistributionMainProcessProcessResponse();
+        response.setJobId("12345");
+        response.setStatusCode("AAA");
+        response.setStatusMessage("Some status message");
+
+        // correct webservice payload
+        StringBuffer sb = new StringBuffer();
+        sb.append("<ns2:DocumentDistributionMainProcessProcessResponse xmlns:ns2=\"http://ag.gov.bc.ca/DocumentDistributionMainProcess\">");
+        sb.append("<ns2:jobId>12345</ns2:jobId>");
+        sb.append("<ns2:statusCode>AAA</ns2:statusCode>");
+        sb.append("<ns2:statusMessage>Some status message</ns2:statusMessage>");
+        sb.append("</ns2:DocumentDistributionMainProcessProcessResponse>");
+        Source source = new StringSource(sb.toString());
+        mockServer.expect(payload(source));
+
+        // no error expected.
+        documentDistributionService.sendResponseToCallback(response);        
+    }
+
+    @Test
+    public void testSendResponseToCallback_invalid() {
+        DocumentDistributionMainProcessProcessResponse response = new DocumentDistributionMainProcessProcessResponse();
+        response.setJobId("12345");
+        response.setStatusCode("AAA");
+        response.setStatusMessage("Some status message");
+
+        // invalid webservice payload (missing namespace)
+        StringBuffer sb = new StringBuffer();
+        sb.append("<ns2:DocumentDistributionMainProcessProcessResponse xmlns:ns2=\"http://ag.gov.bc.ca/DocumentDistributionMainProcess\">");
+        sb.append("<ns2:jobId>12345</ns2:jobId>");
+        sb.append("<ns2:statusCode>AAA</ns2:statusCode>");
+        sb.append("<ns2:statusMessage>Some status message</ns2:statusMessage>");
+        sb.append("</ns2:DocumentDistributionMainProcessProcessResponse>");
+        Source source = new StringSource(sb.toString());
+        mockServer.expect(payload(source)).andRespond(withException(new RuntimeException()));
```

Comment:
q: Is expect methods is the assertion for both valid and invalid test or we can't assert as it's quietly suppressed?

Reply:
Yeah, that's a good question. I was fighting with this for a bit.  Since any exception is quietly suppressed, the only recourse is to parse the logger to confirm the exception was logged.  I looked into a logger interceptor to do this, but couldn't get it work correctly.

Reply:
Thanks @KDNeufeld.